msal/src/main/java/com/microsoft/identity/client/NativeAuthPublicClientApplicationConfigurationFactory.kt

```diff
@@ -0,0 +1,266 @@
+
+        /**
+         * Tweaked loadConfiguration to serialize json into a [NativeAuthPublicClientApplicationConfiguration]
+         * object.
+         */
+        @WorkerThread
+        private fun loadConfiguration(
+            configStream: InputStream,
+            isDefaultConfiguration: Boolean
+        ): NativeAuthPublicClientApplicationConfiguration {
+            val methodTag = "$TAG:loadConfiguration"
+            val buffer: ByteArray
+            try {
+                buffer = ByteArray(configStream.available())
+                configStream.read(buffer)
+            } catch (e: IOException) {
+                if (isDefaultConfiguration) {
+                    throw IllegalStateException("Unable to open default native auth configuration file.", e)
+                } else {
+                    throw IllegalArgumentException("Unable to open provided native auth configuration file.", e)
+                }
+            } finally {
+                try {
+                    configStream.close()
+                } catch (e: IOException) {
+                    if (isDefaultConfiguration) {
+                        Logger.warn(
+                            methodTag,
+                            "Unable to close default native auth configuration file. " +
+                                "This can cause memory leak."
+                        )
+                    } else {
+                        Logger.warn(
+                            methodTag,
+                            "Unable to close provided native auth configuration file. " +
+                                "This can cause memory leak."
+                        )
+                    }
+                }
+            }
+            val config = String(buffer)
+            val gson = getGsonForLoadingConfiguration()
+            return try {
+                gson.fromJson(config, NativeAuthPublicClientApplicationConfiguration::class.java)
+            } catch (e: Exception) {
+                if (e is InterruptedException) {
+                    Thread.currentThread().interrupt()
+                }
+                throw IllegalArgumentException("Error while processing configuration", e)
+            }
+        }
+
+        // This method doesn't belong here, but in the MsalOAuth2Cache. However, we can't
+        // access NativeOAuth2Strategy there, as it isn't in common
+        private fun createCache(components: IPlatformComponents): MsalOAuth2TokenCache<
+            MicrosoftStsOAuth2Strategy,
+            MicrosoftStsAuthorizationRequest,
+            MicrosoftStsTokenResponse,
+            MicrosoftAccount,
+            MicrosoftRefreshToken
+            > {
+            val methodName = ":createCache"
+            Logger.verbose(
+                TAG + methodName,
+                "Creating MsalOAuth2TokenCache"
+            )
+            return MsalOAuth2TokenCache.create(components)
+        }
+
+        private fun getGsonForLoadingConfiguration(): Gson {
```

Comment:
If such a method already exists in the codebase then can you just leverage that?